### PR TITLE
GitHub ci

### PR DESCRIPTION
**Pull request by Michiel-s**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,8 +10,6 @@ cabal.sandbox.config
 /dist-newstyle
 .idea
 
-stack.yaml.lock
```
Comment (hanjoosten) [inline .gitignore]: Why did you remove this? stack.yaml.lock is a generateded file. IMHO it should not be part of version management.

Comment (Michiel-s) [reply .gitignore]: The reason I did this if because I thought it was unintentionally in the gitignore.
It is very common (and also strongly advised) to include lock files in source control. Because than you can really transfer an environment to other developers. Let me explain that quickly.

Scenario **without** lock file in git
Say you have downloaded the source and run stack install months ago locally. It created a stack.yaml.lock. When you now run stack install again (and you didn't update the stack.yaml file) your dependencies are **not** updated. That's ok.
But when I at the same time do a clean clone of the repo, for me the dependencies do get updated, because I don't have a lock file (yet).
So, we are on the same commit, but have different (minor) versions of the dependencies. That is a situation you don't want.

Scenario **with** lock file in git
When you want to update the dependencies, you run `stack update` or so, the lock file also gets updated and committed to git. We both have the same dependencies.

Comment (hanjoosten) [reply .gitignore]: I understand that for build systems with another design philosophy than stack. stack guarantees that the exact same versions are used. The basis for this is the resolver statement in stack.yaml. It points to a set of specific versions of packages. Having the exact same build is one of the key principles in the design of stack. 

The guys of stack know what they are doing. Also, have a look at [their own .gitignore](https://github.com/commercialhaskell/stack/blob/master/.gitignore). It shows my point.

Comment (Michiel-s) [reply .gitignore]: Then I don’t get the point of the lock file. But I thrust you and will add it back to the gitignore.

Comment (hanjoosten) [reply .gitignore]: Thanks for resolving. I [posted your questions on Haskell-stack](https://groups.google.com/forum/#!topic/haskell-stack/xeTN_3UNwE4). Let's see what the pro's have to say about this :)
